Makefile

```diff
@@ -187,19 +187,36 @@ set-manifest-image:
 ## --------------------------------------
 
 RELEASE_TAG := $(shell git describe --abbrev=0 2>/dev/null)
+RELEASE_GOOS := linux darwin
+RELEASE_DIR := out
+RELEASE_BINARIES := clusterctl
+
+$(RELEASE_DIR):
+	mkdir -p $(RELEASE_DIR)/
 
 .PHONY: release
-release:  ## Builds and push container images using the latest git tag for the commit.
+release: clean-release ## Builds and push container images using the latest git tag for the commit.
 	@if [ -z "${RELEASE_TAG}" ]; then echo "RELEASE_TAG is not set"; exit 1; fi
 	# Push the release image to the staging bucket first.
 	REGISTRY=$(STAGING_REGISTRY) TAG=$(RELEASE_TAG) \
 		$(MAKE) docker-build-all docker-push-all
 	# Set the manifest image to the production bucket.
 	MANIFEST_IMG=$(PROD_REGISTRY)/$(IMAGE_NAME) MANIFEST_TAG=$(RELEASE_TAG) \
 		$(MAKE) set-manifest-image
-	# Generate release artifacts.
-	mkdir out/
-	kustomize build config/default > out/cluster-api-components.yaml
+	$(MAKE) release-manifests
+	$(MAKE) release-binaries
+
+.PHONY: release-manifests
+release-manifests: $(RELEASE_DIR) ## Builds the manifests to publish with a release
+	kustomize build config/default > $(RELEASE_DIR)/cluster-api-components.yaml
+
+.PHONY: release-binaries
+release-binaries: $(addprefix release-binary-,$(RELEASE_BINARIES)) ## Builds the binaries to publish with a release
+
+release-binary-%: $(RELEASE_DIR)
+	@for os in $(RELEASE_GOOS); do \
+		CGO_ENABLED=0 GOOS=$${os} GOARCH=amd64 go build -a -ldflags '-extldflags "-static"' -o $(RELEASE_DIR)/$*-$${os}-amd64; \
```

Comment:
Should we build using docker so we're not at the mercy of your locally installed go?

Reply:
Do you have an example? Or can we do this later?

Reply:
Most/all of this is pre go modules, but here's how we set up velero: https://github.com/heptio/velero/blob/29f29f9d614f396dbb310423b65ae16ae129f451/Makefile#L95-L130.

My preference is to only merge a PR that builds in docker. I'm happy to take this over if you want.

Reply:
Added the build in docker command, ptal :)